routes/routes.go

```diff
@@ -1,23 +1,20 @@
 package routes
 
 import (
-	"log"
 	"net/http"
 
 	"github.com/JoaoGumiero/Crud/handlers"
-	"github.com/gorilla/mux"
 )
 
-func UploadRoutes() {
-	r := mux.NewRouter()
-	r.HandleFunc("/", handlers.Index)
-	// Define your HTTP routes using the router
-	r.HandleFunc("/newProduct", handlers.NewProductHandler)
-	r.HandleFunc("/insert", handlers.InsertProductHandler)
-	r.HandleFunc("/delete", handlers.DeleteProductHandler)
-	r.HandleFunc("/edit", handlers.EditProductHandler)
-	r.HandleFunc("/update", handlers.UpdateProductHandler)
+func UploadRoutes() *http.ServeMux {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", handlers.Index)
 
-	log.Println("Server listening on:8000")
-	http.ListenAndServe(":8000", r)
+	mux.HandleFunc("/newProduct", handlers.NewProductHandler)
+	mux.HandleFunc("POST /products", handlers.InsertProductHandler)
+	mux.HandleFunc("DELETE /products", handlers.DeleteProductHandler)
+	mux.HandleFunc("PATCH /products", handlers.EditProductHandler)
+	mux.HandleFunc("PUT /products", handlers.UpdateProductHandler)
```

Comment:
For example here. Your main entity are **Products**, and even playing around with different actions upon them, the route that will be called is always the same: **/products**, the only thing that changes is the HTTP method, as explained in the table below:

| HTTP Method | Route | Description |
| --- | --- | --- |
| GET | /products | List all products |
| POST | /products | Create a new product |
| GET | /products/:id | Get a product |
| PUT | /products/:id | Update a product |
| DELETE | /products/:id | Delete a product |
| PATCH | /products/:id | Partially update a product |

Reply:
And I also saw that you are using some Path values to change things about the product, like: `/products/{id}` and other values, **like name, description, and so on**

I think that for the POST method, there is a better way to do that. But I will give you the pleasure of discovery it kkkk

Reply:
And I also saw that you are using some Path values to change things about the product, like: `/products/{id}` and other values, **like name, description, and so on**

I think that for the POST method, there is a better way to do that. But I will give you the pleasure of discovery it kkkk

Reply:
Oh, I'll take a look at your document right away. I owe you one, tks.